tests/raftstore/cluster.rs

```diff
@@ -146,19 +146,39 @@ impl<T: Simulator> Cluster<T> {
         self.sim.rl().call_command(request, timeout)
     }
 
+    fn call_command_on_leader_once(&mut self,
+                                   region_id: u64,
+                                   mut request: RaftCmdRequest,
+                                   timeout: Duration)
+                                   -> Result<RaftCmdResponse> {
+        if let Some(leader) = self.leader_of_region(region_id) {
+            request.mut_header().set_peer(leader);
+            return self.call_command(request, timeout);
+        }
+        Err(box_err!("can't get leader of region"))
+    }
+
     pub fn call_command_on_leader(&mut self,
                                   region_id: u64,
-                                  mut request: RaftCmdRequest,
+                                  request: RaftCmdRequest,
                                   timeout: Duration)
                                   -> Result<RaftCmdResponse> {
-        for _ in 0..200 {
-            if let Some(leader) = self.leader_of_region(region_id) {
-                request.mut_header().set_peer(leader);
-                return self.call_command(request, timeout);
+        let mut retry_cnt = 0;
+        loop {
+            let result = self.call_command_on_leader_once(region_id, request.clone(), timeout);
+            if let Ok(resp) = result {
```

Comment:
previously `match` was used, and changed to `if let`which was suggested by @ngaut 
...now, I've give up trying to make it clearer after several attempts